src/client/PyCallbacks.cpp

```diff
@@ -18,32 +18,29 @@
  */
 
 #include "PyCallbacks.h"
+#include "utils.h"
 
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
 namespace py = pybind11;
 
-bool PyCallbacks::isHiddenAttribute(const RuleFileInfoUPtr& ruleFileInfo, const wchar_t* key) {
-	for (size_t ai = 0, numAttrs = ruleFileInfo->getNumAttributes(); ai < numAttrs; ai++) {
-		const auto attr = ruleFileInfo->getAttribute(ai);
-		if (std::wcscmp(key, attr->getName()) == 0) {
-			for (size_t k = 0, numAnns = attr->getNumAnnotations(); k < numAnns; k++) {
-				if (std::wcscmp(attr->getAnnotation(k)->getName(), L"@Hidden") == 0)
-					return true;
-			}
-			return false;
-		}
+template <typename T>
+prt::Status PyCallbacks::storeAttr(size_t isIndex, const wchar_t* key, const T value) {
+	if (!isHiddenAttribute(key)) {
+		py::object pyKey = py::cast(pcu::removeDefaultStyleName(key));
+		mModels[isIndex].mAttrVal[pyKey] = value;
 	}
-	return false;
+
+	return prt::STATUS_OK;
 }
 
-std::wstring PyCallbacks::removeDefaultStyleName(const wchar_t* key) {
-	const std::wstring keyName = key;
-	if (keyName.find(L"Default$") == 0)
-		return keyName.substr(8);
-	else
-		return keyName;
+bool PyCallbacks::isHiddenAttribute(const wchar_t* key) {

```

Comment:
add a nullptr check for key... we don't know what those encoder authors do in general ;-)